common/domain/failover_watcher.go

```diff
@@ -183,13 +184,12 @@ func CleanPendingActiveState(
 	// this call has to be made
 	metadata, err := domainManager.GetMetadata(context.Background())
 	if err != nil {
-		return err
+		return fmt.Errorf("getting metadata: %w", err)
```

Comment:
I don't have a great picture in my head for what might be listening to this error. Is there anything that might be doing an exact match on this which needs to be changed to use `errors.As` ?

Reply:
good point. but no, this error is just being used for logging purposes in `coordinator.go:309` and `failower_watcher.go:160`

Reply:
sounds good then